src/backend/distributed/operations/repair_shards.c

```diff
@@ -1199,23 +1204,50 @@ CopyShardTablesViaBlockWrites(List *shardIntervalList, char *sourceNodeName,
 		char *tableOwner = TableOwner(shardInterval->relationId);
 		SendCommandListToWorkerOutsideTransaction(targetNodeName, targetNodePort,
 												  tableOwner, ddlCommandList);
+	}
 
-		ddlCommandList = NIL;
-
+	int taskId = 0;
+	List *copyTaskList = NIL;
+	foreach_ptr(shardInterval, shardIntervalList)
+	{
 		/*
 		 * Skip copying data for partitioned tables, because they contain no
 		 * data themselves. Their partitions do contain data, but those are
 		 * different colocated shards that will be copied seperately.
 		 */
 		if (!PartitionedTable(shardInterval->relationId))
 		{
-			ddlCommandList = CopyShardContentsCommandList(shardInterval, sourceNodeName,
-														  sourceNodePort);
+			StringInfo splitCopyUdfCommand = CreateSplitCopyCommand(
```

Comment:
CreateSplitCopyCommand is not a great name anymore if we are using it for move.

Reply:
Looking more into the PR, since we have to make changes to worker_split_copy_udf as well for things to work, it would be nicer if we kept things seperate.